Rake task for async reindexing of works

awead opened the pull request:
Updates our existing rake task for reindexing works to operate async, putting a job on the queue for each work. This requires refactoring our update_index methods so that committing can be specified at each point along the way.

For each *work* that is indexed, commits are not performed until the the job for the last work is put on the queue.

Fixes #706

mtribone commented:
<img width="1263" alt="screen shot 2017-03-17 at 3 01 00 pm" src="https://cloud.githubusercontent.com/assets/4163828/24058797/e8d6458a-0b22-11e7-8bd9-316ed5e3458a.png">

mtribone commented:
<img width="1259" alt="screen shot 2017-03-17 at 3 01 19 pm" src="https://cloud.githubusercontent.com/assets/4163828/24058828/f83dd42a-0b22-11e7-8c20-3d132f7e3c00.png">

mtribone commented:
<img width="1259" alt="screen shot 2017-03-17 at 3 01 55 pm" src="https://cloud.githubusercontent.com/assets/4163828/24058841/0229f0e0-0b23-11e7-97e8-8370f87aa643.png">

cam156 commented on app/views/curation_concerns/base/_guts4form.html.erb:
```diff
@@ -0,0 +1,47 @@
+<% # we will yield to content_for for each tab, e.g. :files_tab %>
```
@mtribone Did we need this entire override?  Was this to just change the tab name?

mtribone replied:
It was to remove the icons. I may be able to use some CSS trickery to hide them, but I was able to remove them everyplace else except for the main form.

awead committed:
Rake task for async reindexing of works

Updates our existing rake task for reindexing works to operate async,
putting a job on the queue for each work. This requires refactoring our
update_index methods so that committing can be specified at each point
along the way.

For each *work* that is indexed, commits are not performed until the
the job for the last work is put on the queue.
